homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -92,26 +64,16 @@ class MinecraftServerConfigFlow(ConfigFlow, domain=DOMAIN):
                     # Build unique_id and config entry title.
                     unique_id = ""
                     title = f"{host}:{port}"
-                    if ip_address is not None:
-                        # Since IP addresses can change and therefore are not allowed in a
-                        # unique_id, fall back to the MAC address and port (to support
-                        # servers with same MAC address but different ports).
-                        unique_id = f"{mac_address}-{port}"
-                        if ip_address.version == 6:
-                            title = f"[{host}]:{port}"
+                    # Check if 'host' is a valid SRV record.
+                    srv_record = await helpers.async_check_srv_record(self.hass, host)
+                    if srv_record is not None:
+                        # Use only SRV host name in unique_id (does not change).
+                        unique_id = f"{host}-srv"
+                        title = host
                     else:
-                        # Check if 'host' is a valid SRV record.
-                        srv_record = await helpers.async_check_srv_record(
-                            self.hass, host
-                        )
-                        if srv_record is not None:
-                            # Use only SRV host name in unique_id (does not change).
-                            unique_id = f"{host}-srv"
-                            title = host
-                        else:
-                            # Use host name and port in unique_id (to support servers with
-                            # same host name but different ports).
-                            unique_id = f"{host}-{port}"
+                        # Use host name and port in unique_id (to support servers with
+                        # same host name but different ports).
+                        unique_id = f"{host}-{port}"
```

Comment:
What kind of host address is this? Could it change?

Reply:
The host is specified in the configuration, it may be an IPv4/IPv6 address or a DNS name. For example: `minecraft.example.com` or `1.1.1.1`.

Reply:
We normally don't allow IP addresses as unique id as they may change for the same device or service.

Reply:
That is why in the initial PR for the integration I used the MAC address as the unique ID in case of IP addresses (with the limitation that it doesn't work if the MAC address is not readable).

Instead, a second step in the config flow could be added for manual entry of the MAC address, if it's not readable...

Reply:
Found the mentioned review finding:
https://github.com/home-assistant/core/pull/30992#discussion_r373869101

Reply:
If your argument is that the IP address may change, the same could happen for a MAC address. And that a different IP could easily be considered a different server. MAC addresses are also shared between different IPs on a single host, causing a possibility for conflicts.

I'd also point out that for the identifier to change, the server would have to be manually removed and re-added by the user. Servers accessed via a DNS record (weather A, AAAA, or SRV) use this record as part of the identifier (this functionality has not changed).

Reply:
@balloob?

Reply:
No IP addresses.

Reply:
Ok, do you have an alternate suggestion for fixing this bug?

Reply:
As I said before: One solution would be to ask the user for the MAC address, in case it can‘t be read. This can be probably done with a 2nd step in the config flow.

Reply:
> We normally don't allow IP addresses as unique id as they may change for the same device or service.

If the case that the IP address changes the integration would have to be removed and set up again anyway since the Minecraft server would no longer exist on the IP address that Home Assistant would be querying anyway - Because of this I can't see any problem with using the IP address as the entity id? Failing that being allowed I can't see any other attribute that could be the unique ID, aside from the MAC address - which breaks under reasons given above.

Reply:
What if I were to have two Minecraft servers on the same host, and therefore same MAC address?  
Until very recently this was the case for me; I had one server on the default port (25565) and another on 25566. From what I can see, this integration would not be able to handle that since it's relying on the MAC address as a unique ID.  
As pointed out above, if the IP of the server changed then the integration would break anyway, resulting in the need to configure it from scratch, which would in turn result in a new unique ID for the entities.

Reply:
Surely a MAC address isn't really a unique ID? If I'm running more than one minecraft server on the same host, or using a shared hosting platform with multiple servers per host then more than one minecraft server could share the same MAC address. In this instance, an IP:Port combination would be preferred, as it's generally not possible to run multiple services on the same port on a host. 

Additionally, generally servers have static IP assignments and thus the IP address will be unlikely to change without specific user intervention and even if the IP did change, the integration would have to be reconfigured anyway as it connects to an IP and port in order to read data from the minecraft instance.

Reply:
Good point. We should remove mac address too.

Use config entry ID.